components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+
+							case 'guide':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "15"
+								});
+								out.push({
+									"name": "rgShowcaseConfig[15][0][appid]",
+									"value": settings[i][t]["values"]["appid"]
+								});
+								out.push({
+									"name": "rgShowcaseConfig[15][0][publishedfileid]",
+									"value": settings[i][t]["values"]["publishedfileid"]
+								});
+								break;
+
+							case 'achievements':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "17"
+								});
+								if (settings[i][t]["values"].hasOwnProperty("achievements")) {
+									for (var n = 0; n < 7; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"]["achievements"][n] != undefined) {
+											defaultval = [settings[i][t]["values"]["achievements"][n]["appid"], settings[i][t]["values"]["achievements"][n]["title"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[17][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[17][" + n + "][title]",
+											"value": defaultval[1]
+										});
+									}
+								}
+								break;
+
+								case 'games':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "2"
+									});
+
+									if (settings[i][t]["values"].hasOwnProperty("games")) {
+										showcaseconfig = [true, 4, 2, settings[i][t]["values"]["games"]];
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "16"
+									});
+
+									for (var n = 0; n < 4; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "12"
+									});
+
+									for (var n = 0; n < 5; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+							}
+
+							if (showcaseconfig[0]) {
```

Comment:
Wait, so it's internal variable? It'd be explained if it were to be sent to Steam, but now... :-1: